homeassistant/components/unifiprotect/const.py

```diff
@@ -5,8 +5,14 @@
 from uiprotect.data import ModelType, Version
 
 from homeassistant.const import Platform
+from homeassistant.util.hass_dict import HassKey
 
 DOMAIN = "unifiprotect"
+
+# Typed key for hass.data access - stores auth retries per config entry
+DATA_AUTH_RETRIES: HassKey[dict[str, int]] = HassKey(f"{DOMAIN}_auth_retries")
+# Typed key for discovery state
+DATA_DISCOVERY_STARTED: HassKey[bool] = HassKey(f"{DOMAIN}_discovery")
```

Comment:
Would it not be possible to have a single key, initiated inside `async_setup` (for example with auth_retries as an empty dict, and discovery as False)?